ArduCopter/mode.cpp

```diff
@@ -207,6 +207,10 @@ bool Copter::set_mode(Mode::Number mode, ModeReason reason)
     // return immediately if we are already in the desired mode
     if (mode == flightmode->mode_number()) {
         control_mode_reason = reason;
+        // set yaw rate time constant during autopilot startup
+        if (reason == ModeReason::INITIALISED && mode == Mode::Number::STABILIZE) {
```

Comment:
The initial mode may not be STABILIZE.

Reply:
You could set here instead. https://github.com/ArduPilot/ardupilot/blob/deba0b712be9cb7de9b5ccd36f3671e7ea621f27/ArduCopter/system.cpp#L209

Its just a bit painful to be carrying this one time startup code in the main mode change function.